server/configuration.go

```diff
@@ -56,16 +51,7 @@ func (c *configuration) Clone() *configuration {
 
 // IsValid checks if all needed fields are set.
 func (c *configuration) IsValid(isCloud bool) error {
-	switch {
-	case !c.EnableOAuth && !isCloud: // JWT for on-prem
-		switch {
-		case len(c.APIKey) == 0:
-			return errors.New("please configure APIKey")
-
-		case len(c.APISecret) == 0:
-			return errors.New("please configure APISecret")
-		}
-	case c.EnableOAuth || isCloud: // OAuth for either platform
+	if isCloud { // OAuth for either platform
```

Comment:
We want to just remove this if condition, and have the block run unconditionally

```suggestion
```

Reply:
As a follow up explanation here, every other condition removed in this PR is `OAuthEnabled() &&`, whereas this one is `OAuthEnabled() ||` so we treat it slightly differently.